lib/nxdk/xbe.h

```diff
@@ -0,0 +1,24 @@
+/*
+ * Copyright (c) 2019 Stefan Schmidt
+ *
+ * Licensed under the MIT License
+ */
+
+#ifndef __NXDK_XBE_H__
+#define __NXDK_XBE_H__
+
+#ifdef __cplusplus
+{
+#endif
+
+#ifndef __cplusplus
+#include <stdbool.h>
+#endif
+
+void XGetXbePath (char *path, bool includeFileName);
```

Comment:
As these allocate new memory, the `includeFileName` parameter is no longer needed if we also provide something like `dirname()` and `basename()`:

```c
// void XGetXbePath (char *path, bool includeFileName); 
XGetXbePath(tmp, true);  // D:/default.xbe
XGetXbePath(tmp, false);  // D:/

// void XGetXbePath (char *path); 
XGetXbePath(tmp); // D:/default.xbe
dirname(strdup(tmp)); // D:/
basename(strdup(tmp)); // default.xbe
```

I feel this would be a more powerful API that is more extensible in the future (although somewhat minimal in this instance).

Reply:
I don't follow - what allocates memory?

As for the POSIX functions: While I'd usually agree that they're better than having an additional parameter, my intention was to keep things simple for the constructor (not having to check for failed allocations etc.), as it is run very early. Tbh if you prefer to have the parameter gone, I'd rather move the `strrchr` stuff to the constructor and leave the task of implementing `dirname` and `basename` for later.

Reply:
Sorry, I meant they use newly allocated memory (the function doesn't modify the `XeImageFileName`); my post was still claiming that it did allocate memory itself as I thought that was the case, until I constructed the actual code example (and I forgot to change the first sentence).

> Tbh if you prefer to have the parameter gone, I'd rather move the strrchr stuff to the constructor and leave the task of implementing dirname and basename for later.

Yes, I'd prefer that.

Reply:
Done.